### PR TITLE
Add Bucket Sort

### PR DESCRIPTION
**Pull request by muthuannamalai12**
I have added the code for Bucket Sort in java
Issue Number: #147

### CODE REVIEW

**Comment by muthuannamalai12** [comment]
@tarun26091999 please review this PR and please suggest the required changes.

File: Code/Java/Bucket_Sort.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+85
+99
+11
+Enter number of buckets:
+5
+Array before bucket sort:
+88.0 26.0 77.0 49.0 91.0 62.0 33.0 85.0 99.0 11.0 
+Array after Bucket sort:
+11.0 26.0 33.0 49.0 62.0 77.0 85.0 88.0 91.0 99.0
+Time Complexity:  O(n2)
```
Comment (raksha009) [inline Code/Java/Bucket_Sort.java:105]: Plz add a new line above.

Comment (raksha009) [inline Code/Java/Bucket_Sort.java:105]: Write like n^2

File: Code/Java/Bucket_Sort.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        }
+
+        //dividing numbers in different buckets
+        for (int i=0; i<length; i++) {
+            int bucketIndex = (int) ((arr[i] * k) / max_val);
+            bucketList.get(bucketIndex).add(arr[i]);
+        }
+
+        // sorting each bucket one by one
+        for (int i=0; i<k; i++) {
```
Comment (raksha009) [inline Code/Java/Bucket_Sort.java:39]: Add changes around operator: like i = 0; i  < length; i++

File: Code/Java/Bucket_Sort.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+        // after sorting
+        System.out.println("Array after Bucket sort:");
+        for (int i=0; i<arr.length; i++) {
+            System.out.print(arr[i] + " ");
+        }
+        sc.close();
+    }
+}
+/*
```
Comment (raksha009) [inline Code/Java/Bucket_Sort.java:84]: Add a new line

File: Code/Java/Bucket_Sort.java [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+Sample input/output:-
```
Comment (raksha009) [inline Code/Java/Bucket_Sort.java:85]: Format it like Testcases

Input
Output

File: Code/Java/Bucket_Sort.java (hunk 1 above)
Comment (raksha009) [inline Code/Java/Bucket_Sort.java:105]: Add a new line before Time Complexity.

**Review by raksha009 — CHANGES_REQUESTED** [review]

File: Code/Java/Bucket_Sort.java [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+class BucketSort {
+
+    private static void bucketSort(float[] arr, int k) {
+        if (arr.length < 2)
+            return;
+        
+        // getting upper limit for dividing numbers in buckets
+        int length = arr.length;
+        float max_val = arr[0];
+        for (int i = 1; i<length; i++) {
```
Comment (raksha009) [inline Code/Java/Bucket_Sort.java:21]: Add space around the operators like <, =, > etc

File: Code/Java/Bucket_Sort.java [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+            max_val = Math.max(max_val, arr[i]);
+        }
+        max_val += 1;
+
+        //making bucket list and adding buckets in it
+        ArrayList<ArrayList<Float>> bucketList = new ArrayList<ArrayList<Float>>();
+        for (int i=0; i<k; i++) {
```
Comment (raksha009) [inline Code/Java/Bucket_Sort.java:28]: Add space around the operators like <, =, > etc

File: Code/Java/Bucket_Sort.java [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+        System.out.println("Array after Bucket sort:");
+        for (int i=0; i<arr.length; i++) {
+            System.out.print(arr[i] + " ");
+        }
+        sc.close();
+    }
+}
+
+/*
+Test Cases:
```
Comment (raksha009) [inline Code/Java/Bucket_Sort.java:86]: Add 2 test cases

File: Code/Java/Bucket_Sort.java [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+Enter number of buckets:
+5
+
+Output:
+Array before bucket sort:
+88.0 26.0 77.0 49.0 91.0 62.0 33.0 85.0 99.0 11.0 
+Array after Bucket sort:
+11.0 26.0 33.0 49.0 62.0 77.0 85.0 88.0 91.0 99.0
+
+Time Complexity:  O(n^2)
```
Comment (raksha009) [inline Code/Java/Bucket_Sort.java:110]: Mention what is n and k

**Review by raksha009 — CHANGES_REQUESTED** [review]

**Comment by muthuannamalai12** [comment]
@raksha009 I have updated. Kindly review again

**Comment by raksha009** [comment]
Plz go through the guidelines and link yr code to appropriate folder.

**Comment by muthuannamalai12** [comment]
@raksha009 I have linked my code to the appropriate folder.

File: Code/Java/Bucket_Sort.java [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+class BucketSort {
+
+    private static void bucketSort(float[] arr, int k) {
+        if (arr.length < 2)
+            return;
+        
+        // getting upper limit for dividing numbers in buckets
+        int length = arr.length;
+        float max_val = arr[0];
+        for (int i = 1; i<length; i++) {
```
Comment (raksha009) [inline Code/Java/Bucket_Sort.java:21]: Add space around the operators like <
```suggestion
        for (int i = 1; i < length; i++) {
```

File: Code/Java/Bucket_Sort.java [hunk 10]
```diff
@@ -0,0 +1,133 @@
...
+            max_val = Math.max(max_val, arr[i]);
+        }
+        max_val += 1;
+
+        //making bucket list and adding buckets in it
+        ArrayList<ArrayList<Float>> bucketList = new ArrayList<ArrayList<Float>>();
+        for (int i=0; i<k; i++) {
```
Comment (raksha009) [inline Code/Java/Bucket_Sort.java:28]:
```suggestion
        for (int i = 0; i < k; i++) {
```

**Review by raksha009 — CHANGES_REQUESTED** [review]

**Comment by muthuannamalai12** [comment]
@raksha009 I have made the requiered changes

File: Code/Java/Bucket_Sort.java [hunk 11]
```diff
@@ -0,0 +1,133 @@
...
+Enter number of buckets:
+5
+
+Output:
+Array before bucket sort:
+88.0 26.0 77.0 49.0 91.0 62.0 33.0 85.0 99.0 11.0 
+Array after Bucket sort:
+11.0 26.0 33.0 49.0 62.0 77.0 85.0 88.0 91.0 99.0
+
+Example 2:
```
Comment (raksha009) [inline Code/Java/Bucket_Sort.java:110]:
```suggestion
Input:
```

File: Code/Java/Bucket_Sort.java [hunk 12]
```diff
@@ -0,0 +1,133 @@
+/*
+Bucket Sort is a sorting technique that sorts the elements by first dividing the elements into several groups called buckets.
+The elements inside each bucket are sorted using any of the suitable sorting algorithms or recursively calling the same algorithm.
+Several buckets are created. Each bucket is filled with a specific range of elements. 
+The elements inside the bucket are sorted using any other algorithm. Finally, the elements of the bucket are gathered to get the sorted array.
+
```
Comment (raksha009) [inline Code/Java/Bucket_Sort.java:6]: Remove this extra line

File: Code/Java/Bucket_Sort.java [hunk 13]
```diff
@@ -0,0 +1,133 @@
...
+*/
+
+
+import java.util.*;
+
+class BucketSort {
+
```
Comment (raksha009) [inline Code/Java/Bucket_Sort.java:13]: Remove extra line

**Review by raksha009 — CHANGES_REQUESTED** [review]

**Comment by muthuannamalai12** [comment]
@raksha009 I have updated. Kindly review again

**Review by raksha009 — APPROVED** [review]
LBTM

**Review by rudrakshi99 — APPROVED** [review]

**Pull request merged**